### ISSUE TITLE
vrnetlab nodes without their own `kind` in Containerlab

### ISSUE THREAD

**Issue by kaelemc** [srl-labs/containerlab#2335]
This issue is to track vrnetlab nodes which don't have their own kind in Containerlab. Which means there isn't functionality like interface aliases, SSH config, correct hostname etc.

I assume this list will grow.. so far I've added some Cisco nodes. Slowly over time these items should be ticked off as they get added.

- [ ] Cisco ASAv (`asav`) -- this node need some work done on the vrnetlab side.
- [ ] Cisco IOSv (`vios`)
- [ ] Cisco NX-OS (`nxos`)

**Comment by hellt** [comment]
I wouldn't want us to support legacy sysytems as much. If nxos is done and there is a n9kv substitute, do we need it?

Regarding vios, people were using it when this system was added to vrnetlab. What kind did they use?

**Comment by kaelemc** [comment]
Yeah I somewhat agree.. My rationale is it's such a trivial thing to implement for fairly decent UX gains, although it's certainly not a priority.

NX-OS is like XRv. While they both are running older versions of the NOS they're both wayyyyy lighter and have roughly half the resource requirement + half the boot times as their 9k equivalents (n9kv and XRv9k)..

I don't know specifics about NX-OS vs N9Kv but with XRv it supports things like VPLS control-plane which XRv9k doesn't even support.

vIOS is only in your fork of [vrnetlab](https://github.com/hellt/vrnetlab/pull/249). It uses the `linux` kind, but honestly I don't really know how many people are using it since around the same time we released IOL which seemed to interest people a lot more since IOL is not only lighter but it runs IOS-XE code where vIOS is running IOS.. which is pretty much deprecated by Cisco at this point. (and IOS-XE and IOS are basically the same CLI, XE just runs ontop of a linux kernel).

Let me know your thoughts.

**Comment by hellt** [comment]
Ok, I am convinced :) let's do that

**Comment by hellt** [comment]
I think asav has been added already. NX-OS is not a prio, so I will close this
